crypto/evp/ctrl_params_translate.c

```diff
@@ -2274,6 +2300,13 @@ static const struct translation_st evp_pkey_ctx_translations[] = {
     { GET, -1, -1, EVP_PKEY_OP_TYPE_SIG,
       EVP_PKEY_CTRL_GET_MD, NULL, NULL,
       OSSL_SIGNATURE_PARAM_DIGEST, OSSL_PARAM_UTF8_STRING, fix_md },
+
+    /*-
+     * ECX
+     * ===
+     */
+    { SET, -1, -1, EVP_PKEY_OP_KEYGEN, -1, NULL, NULL,
```

Comment:
I am not quite sure it is right to use the wildcard -1 key type here. It will probably work, but just by accident it won't break the EC keygen group support because it is last in the table so the
```
    { SET, EVP_PKEY_EC, 0, EVP_PKEY_OP_PARAMGEN | EVP_PKEY_OP_KEYGEN,
      EVP_PKEY_CTRL_EC_PARAMGEN_CURVE_NID, "ec_paramgen_curve", NULL,
      OSSL_PKEY_PARAM_GROUP_NAME, OSSL_PARAM_UTF8_STRING,
      fix_ec_paramgen_curve_nid },
```
entry is before. I'd suggest using  the EVP_PKEY_X25519, EVP_PKEY_ED25519 as the key types. And add another entry for the X448, ED448 types.

Reply:
@levite might also comment.

Reply:
Isn't this just for X25519 and X448?  At least the fixing function will not work for ED25519 and ED448.  At least looking at https://www.openssl.org/docs/manmaster/man7/EVP_PKEY-X25519.html it seems to be the X variants and not the ED variants.

Reply:
Updated to use the X25519 and X448 key types.